code/messaging/src/main/java/com/adobe/marketing/mobile/FeedItem.java

```diff
@@ -0,0 +1,152 @@
+    private String actionTitle;
+    // Represents when this feed item went live. Represented in seconds since January 1, 1970
+    private long publishedDate;
+    // Represents when this feed item expires. Represented in seconds since January 1, 1970
+    private long expiryDate;
+    // Contains additional key-value pairs associated with this feed item
+    private Map<String, Object> meta;
+
+    /**
+     * Private constructor.
+     * <p>
+     * Use {@link Builder} to create {@link FeedItem} object.
+     */
+    private FeedItem() {}
+
+    /**
+     * {@code FeedItem} Builder.
+     */
+    public static class Builder {
+        private boolean didBuild;
+        final private FeedItem feedItem;
+
+        /**
+         * Builder constructor with required {@code FeedItem} attributes as parameters.
+         * <p>
+         * It sets default values for the remaining {@link FeedItem} attributes.
+         *
+         * @param title required {@link String} plain-text title for the feed item
+         * @param body required {@link String} plain-text body representing the content for the feed item
+         * @param publishedDate required {@code long} represents when this feed item went live
+         * @param expiryDate required {@code long} represents when this feed item expires
+         */
+        Builder(final String title, final String body, final long publishedDate, final long expiryDate) {
+            feedItem = new FeedItem();
+            feedItem.title = StringUtils.isNullOrEmpty(title) ? "" : title;
+            feedItem.body = StringUtils.isNullOrEmpty(body) ? "" : body;
+            feedItem.publishedDate = publishedDate <= 0 ? 0 : publishedDate;
+            feedItem.expiryDate = expiryDate <= 0 ? 0 : expiryDate;
+            feedItem.imageUrl = "";
+            feedItem.actionUrl = "";
+            feedItem.actionTitle = "";
+            feedItem.meta = new HashMap<>();
+            didBuild = false;
+        }
+
+        public Builder setImageUrl(final String imageUrl) {
+            throwIfAlreadyBuilt();
+
+            feedItem.imageUrl = imageUrl;
+            return this;
+        }
+
+        public Builder setActionUrl(final String actionUrl) {
+            throwIfAlreadyBuilt();
+
+            feedItem.actionUrl = actionUrl;
+            return this;
+        }
+
+        public Builder setActionTitle(final String actionTitle) {
+            throwIfAlreadyBuilt();
+
+            feedItem.actionTitle = actionTitle;
+            return this;
+        }
+
+        public Builder setMeta(final Map<String, Object> meta) {
+            throwIfAlreadyBuilt();
+
+            feedItem.meta = meta;
+            return this;
+        }
+
+        public FeedItem build() {
+            if (StringUtils.isNullOrEmpty(feedItem.title) || StringUtils.isNullOrEmpty(feedItem.body) || feedItem.publishedDate <= 0 || feedItem.expiryDate <= 0) {
```

Comment:
nit: I believe you do not need this check here during object construction. You can do this validation when you instantiate a FeedItem object using the provided event data to ensure required fields are present. You can also pinpoint the specific field that's missing as you parse it.

Reply:
the current behavior matches the iOS side which will return a nil if the FeedItem data is missing a required parameter when decoding. i'll keep it like this for now but can update in the future if needed.